packages/mock-block-dock/src/useMockDatastore.ts

```diff
@@ -44,16 +46,38 @@ export const useMockDatastore: UseMockDataStore = (
     entities: [],
     links: [],
     entityTypes: [],
+    blockEntityId: "",
   },
 ) => {
   const [entities, setEntities] = useState<MockDataStore["entities"]>(
     initialData.entities,
   );
+  const [latestBlockEntity, setLatestBlockEntity] = useState(
+    entities.find((entity) => entity.entityId === initialData.blockEntityId),
+  );
   const [links, setLinks] = useState<MockDataStore["links"]>(initialData.links);
   const [entityTypes, _setEntityTypes] = useState<MockDataStore["entityTypes"]>(
     initialData.entityTypes,
   );
 
+  useEffect(() => {
+    setEntities(initialData.entities);
+  }, [initialData.entities]);
+
+  useEffect(() => {
+    let nextLatestBlockEntity = entities.find(
+      (entity) => entity.entityId === initialData.blockEntityId,
+    );
+
+    if (!nextLatestBlockEntity) {
+      nextLatestBlockEntity = initialData.entities.find(
+        (entity) => entity.entityId === initialData.blockEntityId,
+      );
+    }
+
+    setLatestBlockEntity(nextLatestBlockEntity);
```

Comment:
Update `latestBlockEntity` whenever either initialData entities, or the internal entities state changes.

Reply:
`latestBlockEntity` is derived from the props and state so I don't think it needs to be in state itself - it can just be recalculated on each render (or inside `useMemo`)